### PR TITLE
Fix eager loading of scoped associations. Previously the scope was no…

### PR DESCRIPTION
**Pull request by asedge**
…t applied.  Scopes that accept arguments cannot be used as they are expecting an instance to be passed to them and if we are eager loading there is no instance to reference.

### CODE REVIEW

File: lib/active_force/association/eager_load_projection_builder.rb [hunk 1]
```diff
@@ -43,17 +51,17 @@ class HasManyAssociationProjectionBuilder < AbstractProjectionBuilder
       # to be pluralized
       def projections
         relationship_name = association.sfdc_association_field
-        query = Query.new relationship_name
+        query = ActiveQuery.new(association.relation_model, relationship_name)
```
Comment (bfrey08) [inline lib/active_force/association/eager_load_projection_builder.rb:54]: Asking for my own understanding, why the change from Query to ActiveQuery here?

Comment (bfrey08) [reply lib/active_force/association/eager_load_projection_builder.rb:54]: Ah wait, you needed ActiveQuery for the association. Disregard

File: lib/active_force/association/eager_load_projection_builder.rb [hunk 2]
```diff
@@ -34,6 +35,13 @@ def initialize(association, parent_association_field = nil)
       def projections
         raise "Must define #{self.class.name}#projections"
       end
+
+      def apply_association_scope(query)
+        return query unless association.scoped?
+        raise InvalidEagerLoadAssociation, "Cannot use scopes that expect arguments: #{association.relation_name}" if association.scoped_as.arity.positive?
+
+        query.instance_exec(&association.scoped_as)
```
Comment (bfrey08) [inline lib/active_force/association/eager_load_projection_builder.rb:43]: Fancy

**Review by bfrey08 — APPROVED** [review]
LGTM!

File: lib/active_force/association/eager_load_projection_builder.rb (hunk 1 above)
Comment (asedge) [reply lib/active_force/association/eager_load_projection_builder.rb:54]: `ActiveQuery` has and calls `build_condition` which translates the fields to their SF equivalents, plus some more.  Applying the scope without that yields some malformed query errors.

**Review by salonishah23 — COMMENTED** [review]
This def opens up the possibilities of using eager loading correctly. Great work!
